db/crud.go

```diff
@@ -893,6 +905,20 @@ func (db *Database) storeOldBodyInRevTreeAndUpdateCurrent(doc *Document, prevCur
 		if marshalErr != nil {
 			base.WarnfCtx(db.Ctx, base.KeyAll, "Unable to marshal document body for storage in rev tree: %v", marshalErr)
 		}
+		// Stamp _attachments and _deleted into rev tree bodies
+		oldBodyJson, marshalErr = base.InjectJSONProperties(
+			oldBodyJson,
+			base.KVPair{
+				Key: BodyAttachments,
+				Val: doc.SyncData.Attachments,
```

Comment:
As discussed, do an empty check on attachments first to avoid adding empty structs.

Also we don't want to be setting _deleted:false in any cases.